### PR TITLE
fixes some small thread-related issues

### PR DESCRIPTION
**Pull request by sebastianbaginski**
- fixed destroying a locked mutex in `ThreadTest`
- removed race condition on static data generated by rock in `Pointer` class

@marcusnaslund

### CODE REVIEW

File: source/concurrent/SynchronizedQueue.ooc [hunk 1]
```diff
@@ -63,8 +63,8 @@ BlockedQueue: class <T> extends SynchronizedQueue<T> {
 	}
 	cancel: func { this _waitLock with(|| this _canceled = true) . wakeAll() }
 	wait: func -> T {
-		result: T = null
 		this _waitLock lockWhen(func -> Bool { !this empty || this _canceled })
+		result: T = null
```
Comment (sebastianbaginski) [inline source/concurrent/SynchronizedQueue.ooc]: This innocent looking like can actually modify static data in Pointer class:
```
types__PointerClass *types__Pointer_class(){
    static _Bool __done__ = false;
...
__done__ = true;
...
}
```

Comment (marcusnaslund) [inline source/concurrent/SynchronizedQueue.ooc]: What does this have to do with the placement of `result`?

**Comment by marcusnaslund** [comment]
Did you try all tests or just this one?

**Comment by marcusnaslund** [comment]
Oops, sorry.

File: source/concurrent/SynchronizedQueue.ooc (hunk 1 above)
Comment (sebastianbaginski) [inline source/concurrent/SynchronizedQueue.ooc]: Related to C code it generates.
This line was previously outside the critical section:
```C
Memory__memcpy(result, &(__SynchronizedQueue_unreferencable4), ((types__Class*)types__Pointer_class())->size);
```
and `types__Pointer_class()` modifies static data.

Comment (marcusnaslund) [inline source/concurrent/SynchronizedQueue.ooc]: *sad*

**Comment by sebastianbaginski** [comment]
> Did you try all tests or just this one?

There are many more issues, but we need to verify which of them are false positives. For now I wanted to fix only the problems I'm absolutely sure of.

**Comment by marcusnaslund** [comment]
Alright.
We have problems with `PromiseTest` sometimes (especially when running memcheck.sh), there's definitely something wrong there.

**Comment by sebastianbaginski** [comment]
> (...) there's definitely something wrong there

That's quite probable:

`ERROR SUMMARY: 117 errors from 12 contexts (suppressed: 312 from 44)`
